File: examples/python-examples/MeshFixDegeneracies.dox.py
import sys

import meshlib.mrmeshpy as mrmeshpy

# Load mesh
mesh = mrmeshpy.loadMesh("mesh.stl")

# you can set various parameters for the resolving process; see the documentation for more info
params = mrmeshpy.FixMeshDegeneraciesParams()
params.maxDeviation = 1e-5 * mesh.computeBoundingBox().diagonal()
params.tinyEdgeLength = 1e-3

mrmeshpy.fixMeshDegeneracies(mesh, params)

mrmeshpy.saveMesh(mesh, "fixed_mesh.stl")

File: examples/js-examples/MeshFixDegeneracies.dox.js
import createMeshLib from '@meshinspector/meshlib';

const ml = await createMeshLib();

using mesh = ml.MeshLoad.fromAnySupportedFormat('mesh.stl');

// you can set various parameters for the resolving process; see the documentation for more info
using box = mesh.computeBoundingBox();
using params = new ml.FixMeshDegeneraciesParams();
params.maxDeviation = 1e-5 * box.diagonal();
params.tinyEdgeLength = 1e-3;
ml.fixMeshDegeneracies(mesh, params);

// Save result
ml.MeshSave.toAnySupportedFormat(mesh, 'fixedMesh.stl');
